cmat/trait_mapping/ols.py

```diff
@@ -92,7 +92,24 @@ def is_in_efo(uri: str) -> bool:
     Checks whether given ontology uri is a valid term in EFO.
 
     :param uri: Ontology uri to use in querying EFO using OLS
-    :return: Boolean value, true if ontology uri is valid and non-obsolete term in EFO
+    :return: Boolean value, true if ontology uri is valid term in EFO
     """
     response = ols_efo_query(uri)
     return response.status_code == 200
+
+
+@lru_cache(maxsize=16384)
+def get_replacement_term(uri: str) -> str:
+    """
+    Finds replacement term in EFO (if present) for the given ontology uri.
+
+    :param uri: Ontology uri to use in querying EFO using OLS
+    :return: Replacement term URI or empty string if not obsolete
+    """
+    response = ols_efo_query(uri)
+    if response.status_code != 200:
```

Comment:
I am not very familiar with the said URI but I think there can be other reasons for a status response other than 200 (like internal server failure at their end). Since the ols_efo_query does not seem to retry on failure, can this code be unforgiving when there are occasional glitches at the server side?

Reply:
I agree, there are other ways to handle the response from OLS API for missing terms. Nevertheless, AFAIK any term that would be skipped incorrectly this way would be re-parsed in the next iteration, so it would be picked and corrected, I assume

Reply:
Good point on retries, amended `ols_efo_query` to retry on server failures. The only other relevant status code for this API is `404 Not Found`, so I think checking for `200` is still appropriate here.